python/cudf/cudf/core/multiindex.py

```diff
@@ -392,13 +393,124 @@ def _popn(self, n):
         return result
 
     def __repr__(self):
-        return (
-            "MultiIndex(levels="
-            + str(self.levels)
-            + ",\ncodes="
-            + str(self.codes)
-            + ")"
+        max_seq_items = get_option("display.max_seq_items") or len(self)
+
+        if len(self) > max_seq_items:
+            n = int(max_seq_items / 2) + 1
+            indices = list(range(n))
+            indices.extend(list(range(len(self) - n, len(self))))
+            preprocess = self.take(indices)
+        else:
+            preprocess = self
+
+        preprocess = preprocess.to_pandas(nullable=True)
+        output = preprocess.__repr__()
+        cols_nulls = [
+            self._source_data._data[col].has_nulls
+            for col in self._source_data._data
+        ]
+
+        output_prefix = "MultiIndex("
```

Comment:
Let's use `self.__class__.__name__` instead of hard coding the name of the class. This way, a subclass `MyMultiIndex(MultiIndex)` will have the correct name in its repr.

Reply:
Updated this to use the class name attribute